feature_tests/kotlin/somelib/src/main/kotlin/dev/diplomattest/somelib/CallbackWrapper.kt

```diff
@@ -0,0 +1,281 @@
+import com.sun.jna.Pointer
+import com.sun.jna.Structure
+
+internal interface CallbackWrapperLib: Library {
+    fun CallbackWrapper_test_multi_arg_callback(f: DiplomatCallback_CallbackWrapper_test_multi_arg_callback_diplomatCallback_f_Native, x: Int): Int
+    fun CallbackWrapper_test_no_args(h: DiplomatCallback_CallbackWrapper_test_no_args_diplomatCallback_h_Native): Int
+    fun CallbackWrapper_test_cb_with_struct(f: DiplomatCallback_CallbackWrapper_test_cb_with_struct_diplomatCallback_f_Native): Int
+    fun CallbackWrapper_test_multiple_cb_args(f: DiplomatCallback_CallbackWrapper_test_multiple_cb_args_diplomatCallback_f_Native, g: DiplomatCallback_CallbackWrapper_test_multiple_cb_args_diplomatCallback_g_Native): Int
+}
+
+internal class CallbackWrapperNative: Structure(), Structure.ByValue {
+    @JvmField
+    internal var cantBeEmpty: Byte = 0;
+  
+    // Define the fields of the struct
+    override fun getFieldOrder(): List<String> {
+        return listOf("cantBeEmpty")
+    }
+}
+
+
+internal interface Runner_DiplomatCallback_CallbackWrapper_test_multi_arg_callback_diplomatCallback_f: Callback {
+    fun invoke(ignored: Pointer?, arg0: Int ): Int
+}
+
+class DiplomatCallback_CallbackWrapper_test_multi_arg_callback_diplomatCallback_f_Native: Structure(), Structure.ByValue {
+    @JvmField
+    internal var dataa: Pointer = Pointer(0L);
+    @JvmField
+    internal var run_callback: Runner_DiplomatCallback_CallbackWrapper_test_multi_arg_callback_diplomatCallback_f
+        = object :  Runner_DiplomatCallback_CallbackWrapper_test_multi_arg_callback_diplomatCallback_f {
+                override fun invoke(ignored: Pointer?, arg0: Int ): Int {
+                    throw Exception("ERROR NOT IMPLEMENTED")
+                }
+            }
+    @JvmField
+    internal var destructor: Pointer = Pointer(0L);
+
+    // Define the fields of the struct
+    override fun getFieldOrder(): List<String> {
+        return listOf("dataa", "run_callback", "destructor")
+    }
+}
+
+class DiplomatCallback_CallbackWrapper_test_multi_arg_callback_diplomatCallback_f internal constructor (
+    internal val nativeStruct: DiplomatCallback_CallbackWrapper_test_multi_arg_callback_diplomatCallback_f_Native) {
+    val dataa: Pointer = nativeStruct.dataa
+    val run_callback: Callback = nativeStruct.run_callback
+    val destructor: Pointer = nativeStruct.destructor
+
+    companion object {
+        val NATIVESIZE: Long = Native.getNativeSize(DiplomatCallback_CallbackWrapper_test_multi_arg_callback_diplomatCallback_f_Native::class.java).toLong()
+        
+        fun fromCallback(cb: (Int)->Int): DiplomatCallback_CallbackWrapper_test_multi_arg_callback_diplomatCallback_f {
+            val callback: Runner_DiplomatCallback_CallbackWrapper_test_multi_arg_callback_diplomatCallback_f = object :  Runner_DiplomatCallback_CallbackWrapper_test_multi_arg_callback_diplomatCallback_f {
+                override fun invoke(ignored: Pointer?, arg0: Int ): Int {
```

Comment:
question: what's the deal with the "ignored"?

Reply:
Ah, so the `run_callback` signature on the Rust side is `unsafe extern "C" fn(*const c_void, ...) -> ReturnType` -- the first argument is a pointer that could contain language-specific information/context, but for Kotlin none is needed so it's ignored. Maybe it should have a different though?

Reply:
yeah let's call it lang_specific_context or something